### PR TITLE
provider: Support default tags (resources aws_e* 1)

### PR DESCRIPTION
**Pull request by YakDriver**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Reference: https://github.com/hashicorp/terraform-provider-aws/issues/7926

Output from acceptance testing: Will be handled via CI, review, and daily acceptance testing

### CODE REVIEW

File: aws/resource_aws_ebs_snapshot.go [hunk 1]
```diff
@@ -67,17 +69,20 @@ func resourceAwsEbsSnapshot() *schema.Resource {
...
 
 func resourceAwsEbsSnapshotCreate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ec2conn
+	defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
+	tags := defaultTagsConfig.MergeTags(keyvaluetags.New(d.Get("tags").(map[string]interface{})))
 
 	request := &ec2.CreateSnapshotInput{
 		VolumeId:          aws.String(d.Get("volume_id").(string)),
-		TagSpecifications: ec2TagSpecificationsFromMap(d.Get("tags").(map[string]interface{}), ec2.ResourceTypeSnapshot),
+		TagSpecifications: ec2TagSpecificationsFromKeyValueTags(tags, ec2.ResourceTypeSnapshot),
```
Comment (YakDriver) [inline aws/resource_aws_ebs_snapshot.go:85]: Ignore?

File: aws/resource_aws_ebs_volume.go [hunk 2]
```diff
@@ -98,10 +103,12 @@ func resourceAwsEbsVolume() *schema.Resource {
 
 func resourceAwsEbsVolumeCreate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ec2conn
+	defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
+	tags := defaultTagsConfig.MergeTags(keyvaluetags.New(d.Get("tags").(map[string]interface{})))
 
 	request := &ec2.CreateVolumeInput{
 		AvailabilityZone:  aws.String(d.Get("availability_zone").(string)),
-		TagSpecifications: ec2TagSpecificationsFromMap(d.Get("tags").(map[string]interface{}), ec2.ResourceTypeVolume),
+		TagSpecifications: ec2TagSpecificationsFromKeyValueTags(tags, ec2.ResourceTypeVolume),
```
Comment (YakDriver) [inline aws/resource_aws_ebs_volume.go:111]: Ignore?

File: aws/resource_aws_ec2_capacity_reservation.go [hunk 3]
```diff
@@ -121,14 +124,16 @@ func resourceAwsEc2CapacityReservation() *schema.Resource {
...
+	tags := defaultTagsConfig.MergeTags(keyvaluetags.New(d.Get("tags").(map[string]interface{})))
 
 	opts := &ec2.CreateCapacityReservationInput{
 		AvailabilityZone:  aws.String(d.Get("availability_zone").(string)),
 		EndDateType:       aws.String(d.Get("end_date_type").(string)),
 		InstanceCount:     aws.Int64(int64(d.Get("instance_count").(int))),
 		InstancePlatform:  aws.String(d.Get("instance_platform").(string)),
 		InstanceType:      aws.String(d.Get("instance_type").(string)),
-		TagSpecifications: ec2TagSpecificationsFromMap(d.Get("tags").(map[string]interface{}), ec2ResourceTypeCapacityReservation),
+		TagSpecifications: ec2TagSpecificationsFromKeyValueTags(tags, ec2ResourceTypeCapacityReservation),
```
Comment (YakDriver) [inline aws/resource_aws_ec2_capacity_reservation.go:136]: Ignore?

File: aws/resource_aws_ec2_carrier_gateway.go [hunk 4]
```diff
@@ -49,9 +52,11 @@ func resourceAwsEc2CarrierGateway() *schema.Resource {
 
 func resourceAwsEc2CarrierGatewayCreate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ec2conn
+	defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
+	tags := defaultTagsConfig.MergeTags(keyvaluetags.New(d.Get("tags").(map[string]interface{})))
 
 	input := &ec2.CreateCarrierGatewayInput{
-		TagSpecifications: ec2TagSpecificationsFromMap(d.Get("tags").(map[string]interface{}), "carrier-gateway"),
+		TagSpecifications: ec2TagSpecificationsFromKeyValueTags(tags, "carrier-gateway"),
```
Comment (YakDriver) [inline aws/resource_aws_ec2_carrier_gateway.go:59]: Ignore?

File: aws/resource_aws_ec2_client_vpn_endpoint.go [hunk 5]
```diff
@@ -136,13 +139,15 @@ func resourceAwsEc2ClientVpnEndpoint() *schema.Resource {
...
+	defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
+	tags := defaultTagsConfig.MergeTags(keyvaluetags.New(d.Get("tags").(map[string]interface{})))
 
 	req := &ec2.CreateClientVpnEndpointInput{
 		ClientCidrBlock:      aws.String(d.Get("client_cidr_block").(string)),
 		ServerCertificateArn: aws.String(d.Get("server_certificate_arn").(string)),
 		TransportProtocol:    aws.String(d.Get("transport_protocol").(string)),
 		SplitTunnel:          aws.Bool(d.Get("split_tunnel").(bool)),
-		TagSpecifications:    ec2TagSpecificationsFromMap(d.Get("tags").(map[string]interface{}), ec2.ResourceTypeClientVpnEndpoint),
+		TagSpecifications:    ec2TagSpecificationsFromKeyValueTags(tags, ec2.ResourceTypeClientVpnEndpoint),
```
Comment (YakDriver) [inline aws/resource_aws_ec2_client_vpn_endpoint.go:150]: Ignore?

File: aws/resource_aws_ec2_fleet.go [hunk 6]
```diff
@@ -346,7 +351,7 @@ func resourceAwsEc2FleetCreate(d *schema.ResourceData, meta interface{}) error {
 		SpotOptions:                      expandEc2SpotOptionsRequest(d.Get("spot_options").([]interface{})),
 		TargetCapacitySpecification:      expandEc2TargetCapacitySpecificationRequest(d.Get("target_capacity_specification").([]interface{})),
 		TerminateInstancesWithExpiration: aws.Bool(d.Get("terminate_instances_with_expiration").(bool)),
-		TagSpecifications:                ec2TagSpecificationsFromMap(d.Get("tags").(map[string]interface{}), ec2.ResourceTypeFleet),
+		TagSpecifications:                ec2TagSpecificationsFromKeyValueTags(tags, ec2.ResourceTypeFleet),
```
Comment (YakDriver) [inline aws/resource_aws_ec2_fleet.go:354]: Ignore?

File: aws/resource_aws_ec2_local_gateway_route_table_vpc_association.go [hunk 7]
```diff
@@ -48,10 +51,12 @@ func resourceAwsEc2LocalGatewayRouteTableVpcAssociation() *schema.Resource {
 
 func resourceAwsEc2LocalGatewayRouteTableVpcAssociationCreate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ec2conn
+	defaultTagsConfig := meta.(*AWSClient).DefaultTagsConfig
+	tags := defaultTagsConfig.MergeTags(keyvaluetags.New(d.Get("tags").(map[string]interface{})))
 
 	req := &ec2.CreateLocalGatewayRouteTableVpcAssociationInput{
 		LocalGatewayRouteTableId: aws.String(d.Get("local_gateway_route_table_id").(string)),
-		TagSpecifications:        ec2TagSpecificationsFromMap(d.Get("tags").(map[string]interface{}), ec2ResourceTypeLocalGatewayRouteTableVpcAssociation),
+		TagSpecifications:        ec2TagSpecificationsFromKeyValueTags(tags, ec2ResourceTypeLocalGatewayRouteTableVpcAssociation),
```
Comment (YakDriver) [inline aws/resource_aws_ec2_local_gateway_route_table_vpc_association.go:59]: Ignore?

File: aws/resource_aws_ec2_managed_prefix_list.go [hunk 8]
```diff
@@ -111,7 +117,7 @@ func resourceAwsEc2ManagedPrefixListCreate(d *schema.ResourceData, meta interfac
 	}
 
 	if v, ok := d.GetOk("tags"); ok && len(v.(map[string]interface{})) > 0 {
-		input.TagSpecifications = ec2TagSpecificationsFromMap(v.(map[string]interface{}), "prefix-list")
+		input.TagSpecifications = ec2TagSpecificationsFromKeyValueTags(tags, "prefix-list")
```
Comment (YakDriver) [inline aws/resource_aws_ec2_managed_prefix_list.go:120]: Ignore?

File: aws/resource_aws_ebs_volume.go [hunk 9]
```diff
@@ -26,7 +27,10 @@ func resourceAwsEbsVolume() *schema.Resource {
 			State: schema.ImportStatePassthrough,
 		},
 
-		CustomizeDiff: resourceAwsEbsVolumeCustomizeDiff,
+		CustomizeDiff: customdiff.All(
```
Comment (anGie44) [inline aws/resource_aws_ebs_volume.go:30]: similar comment in the `e* 2` PR , `customdiff.Sequence` _may_ be preferred here

File: aws/resource_aws_ec2_managed_prefix_list.go [hunk 10]
```diff
@@ -28,10 +28,13 @@ func resourceAwsEc2ManagedPrefixList() *schema.Resource {
 			State: schema.ImportStatePassthrough,
 		},
 
-		CustomizeDiff: customdiff.Sequence(
-			customdiff.ComputedIf("version", func(ctx context.Context, diff *schema.ResourceDiff, meta interface{}) bool {
-				return diff.HasChange("entry")
-			}),
+		CustomizeDiff: customdiff.All(
+			customdiff.Sequence(
```
Comment (anGie44) [inline aws/resource_aws_ec2_managed_prefix_list.go:32]: similar comment as above

File: website/docs/r/ec2_carrier_gateway.html.markdown [hunk 11]
```diff
@@ -27,13 +27,14 @@ resource "aws_ec2_carrier_gateway" "example" {
...
 * `vpc_id` - (Required) The ID of the VPC to associate with the carrier gateway.
-* `tags` - (Optional) A map of tags to assign to the resource.
+* `tags` - (Optional) A map of tags to assign to the resource. If configured with a provider [`default_tags` configuration block](https://www.terraform.io/docs/providers/aws/index.html#default_tags-configuration-block) present, tags with matching keys will overwrite those defined at the provider-level.
 
 ## Attributes Reference
 
 In addition to all arguments above, the following attributes are exported:
 
 * `id` - The ID of the carrier gateway.
+* `tags_all` - A map of tags assigned to the resource, including those inherited from the provider [`default_tags` configuration block](https://www.terraform.io/docs/providers/aws/index.html#default_tags-configuration-block).
```
Comment (anGie44) [inline website/docs/r/ec2_carrier_gateway.html.markdown:37]: the list isn't already in alphabetical order, but this attribute may sit better if added to the end of the list here

File: website/docs/r/ec2_client_vpn_endpoint.html.markdown [hunk 12]
```diff
@@ -69,6 +69,7 @@ One of the following arguments must be supplied:
 In addition to all arguments above, the following attributes are exported:
 
 * `id` - The ID of the Client VPN endpoint.
+* `tags_all` - A map of tags assigned to the resource, including those inherited from the provider [`default_tags` configuration block](https://www.terraform.io/docs/providers/aws/index.html#default_tags-configuration-block).
```
Comment (anGie44) [inline website/docs/r/ec2_client_vpn_endpoint.html.markdown:72]: similar comment as above: just to shift this attribute to the end of the list

File: website/docs/r/ec2_managed_prefix_list.html.markdown [hunk 13]
```diff
@@ -66,6 +66,7 @@ The `entry` block supports:
 In addition to all arguments above, the following attributes are exported:
 
 * `id` - The ID of the prefix list.
+* `tags_all` - A map of tags assigned to the resource, including those inherited from the provider [`default_tags` configuration block](https://www.terraform.io/docs/providers/aws/index.html#default_tags-configuration-block).
```
Comment (anGie44) [inline website/docs/r/ec2_managed_prefix_list.html.markdown:69]: similar comment as above to shift the placement of this attribute

**Review by anGie44 — APPROVED** [review]
Just some nits in the documentation and the open comment re: `customdiff.Sequence()`, otherwise LGTM

**Pull request merged**